### ISSUE TITLE
[Cream] Crash in train.py

### ISSUE THREAD

**Issue by duchengyao** [microsoft/nni#3418]
1. Unable to load checkpoint. Because we use torch.save and timm.load.
2. [assert teacher_cand is not None](https://github.com/microsoft/nni/blob/59521d330fee49b8d72f983a9c940aa51473cc81/nni/algorithms/nas/pytorch/cream/trainer.py#L131) because `self.current_teacher_arch` is  always None in `self.prioritized_board.append`.  Is [this line](https://github.com/microsoft/nni/blob/master/nni/algorithms/nas/pytorch/cream/trainer.py#L168) should be `self.current_student_arch` ?

**Comment by Z7zuqer** [comment]
Hi,

Q1: Could you show more details about your failure to load checkpoints? We didn't store checkpoints during supernet training. While in retrain stage, we use timm.save and timm.load to deal with the checkpoint. So more details are needed to know the problem.

Q2: No. In [this line](https://github.com/microsoft/nni/blob/59521d330fee49b8d72f983a9c940aa51473cc81/nni/algorithms/nas/pytorch/cream/trainer.py#L131), we only deal with teacher architecture in prioritized board. So if *teacher_cand* is asserted as None, it means failure of training meta networks, probably caused by large learning rate or NaN gradients. You should delicately tune the training parameters to avoid such circumstances.

Best,
Hao.

**Comment by penghouwen** [comment]
@duchengyao Hi, I am wondering whether you have solved the issue? Hao's replies may help you. Pls feel free to contact us. Thanks.

**Comment by duchengyao** [comment]
The problem has been solved. I can run the Cram algorithm in nni. Thank you for your reply. ^_^
